[OpenVINO backend] test_logsumexp fails

In this function:
https://github.com/keras-team/keras/blob/master/keras/src/backend/openvino/math.py#L37-L55 (logsumexp)

Running the math test with openvino backend:
```bash
pytest -c pytest.ini keras/src/ops/math_test.py -v
```
Gives:
```
=================================== FAILURES ===================================
____________________ MathOpsCorrectnessTest.test_logsumexp _____________________

self = <keras.src.ops.math_test.MathOpsCorrectnessTest testMethod=test_logsumexp>

    def test_logsumexp(self):
        x = np.random.rand(5, 5)
        outputs = kmath.logsumexp(x)
        expected = np.log(np.sum(np.exp(x)))
        self.assertAllClose(outputs, expected)
    
        outputs = kmath.logsumexp(x, axis=1)
        expected = np.log(np.sum(np.exp(x), axis=1))
>       self.assertAllClose(outputs, expected)

keras/src/ops/math_test.py:680: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
keras/src/testing/test_case.py:48: in assertAllClose
    np.testing.assert_allclose(x1, x2, atol=atol, rtol=rtol, err_msg=msg)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

args = (<function assert_allclose.<locals>.compare at 0x7572ef871080>, array([2.32454872, 2.03384614, 2.43761063, 1.91885555, 2.03327584]), array([2.23744892, 2.11629929, 2.30870708, 1.99097716, 2.06320653]))
kwds = {'equal_nan': True, 'err_msg': 'None', 'header': 'Not equal to tolerance rtol=1e-06, atol=1e-06', 'strict': False, ...}

    @wraps(func)
    def inner(*args, **kwds):
        with self._recreate_cm():
>           return func(*args, **kwds)
                   ^^^^^^^^^^^^^^^^^^^
E           AssertionError: 
E           Not equal to tolerance rtol=1e-06, atol=1e-06
E           None
E           Mismatched elements: 5 / 5 (100%)
E           Max absolute difference among violations: 0.12890355
E           Max relative difference among violations: 0.05583365
E            ACTUAL: array([2.324549, 2.033846, 2.437611, 1.918856, 2.033276])
E            DESIRED: array([2.237449, 2.116299, 2.308707, 1.990977, 2.063207])

/usr/lib/python3.12/contextlib.py:81: AssertionError
=============================== warnings summary ===============================
../../../../../usr/lib/python3.12/multiprocessing/popen_fork.py:66
../../../../../usr/lib/python3.12/multiprocessing/popen_fork.py:66
  /usr/lib/python3.12/multiprocessing/popen_fork.py:66: DeprecationWarning: This process (pid=10201) is multi-threaded, use of fork() may lead to deadlocks in the child.
    self.pid = os.fork()

../env/lib/python3.12/site-packages/openvino/runtime/__init__.py:10
  /home/mohamed-ashraf/Desktop/GSoC2025/env/lib/python3.12/site-packages/openvino/runtime/__init__.py:10: DeprecationWarning: The `openvino.runtime` module is deprecated and will be removed in the 2026.0 release. Please replace `openvino.runtime` with `openvino`.
    warnings.warn(

../env/lib/python3.12/site-packages/_pytest/config/__init__.py:1474
  /home/mohamed-ashraf/Desktop/GSoC2025/env/lib/python3.12/site-packages/_pytest/config/__init__.py:1474: PytestConfigWarning: Unknown config option: env
  
    self._warn_or_fail_if_strict(f"Unknown config option: {key}\n")

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED keras/src/ops/math_test.py::MathOpsCorrectnessTest::test_logsumexp - A...
============= 1 failed, 74 passed, 59 skipped, 4 warnings in 0.69s =============
```

--- a/keras/src/backend/openvino/excluded_concrete_tests.txt
+++ b/keras/src/backend/openvino/excluded_concrete_tests.txt
@@ -207,7 +207,6 @@ MathOpsCorrectnessTest::test_istft4
 MathOpsCorrectnessTest::test_istft5
 MathOpsCorrectnessTest::test_istft6
 MathOpsCorrectnessTest::test_logdet
-MathOpsCorrectnessTest::test_logsumexp
 MathOpsCorrectnessTest::test_rfft0
 MathOpsCorrectnessTest::test_rfft1
 MathOpsCorrectnessTest::test_rfft2

--- a/keras/src/backend/openvino/math.py
+++ b/keras/src/backend/openvino/math.py
@@ -44,13 +44,17 @@ def logsumexp(x, axis=None, keepdims=False):
         axis = list(axis)
     axis = ov_opset.constant(axis, Type.i32).output(0)
     const_zero = ov_opset.constant(0, x.get_element_type()).output(0)
-    reduce_max = ov_opset.reduce_max(x, axis, keepdims).output(0)
+    # Use keepdims=True for reduce_max to ensure proper broadcasting
+    reduce_max = ov_opset.reduce_max(x, axis, True).output(0)
     is_finite = ov_opset.is_finite(reduce_max).output(0)
     norm_max = ov_opset.select(is_finite, reduce_max, const_zero).output(0)
     norm_max_sub = ov_opset.subtract(x, norm_max).output(0)
     exp_norm_max = ov_opset.exp(norm_max_sub).output(0)
     sum_exp = ov_opset.reduce_sum(exp_norm_max, axis, keepdims).output(0)
     log_sum_exp = ov_opset.log(sum_exp).output(0)
+    # Squeeze norm_max if needed to match dimensions
+    if not keepdims:
+        norm_max = ov_opset.squeeze(norm_max, axis).output(0)
     log_sum_exp = ov_opset.add(norm_max, log_sum_exp).output(0)
     return OpenVINOKerasTensor(log_sum_exp)
 

